### PR TITLE
exercise complete

### PR DESCRIPTION
**Pull request by mystioreo**
time complexity: O(n^2) because of nester loops
space complexity: O(1) because the string is reversed in place

### CODE REVIEW

**Comment by shrutivanw** [comment]
Nice work!

Minor note on the code: Since you're doing all the manipulations in-place, you don't need to return anything. The return statements on lines 6, 7 and 40 could simply be `return`.

The space complexity is correct: it is constant.
The time complexity for your algorithm will be O(n) or linear. The outer loop determines the beginning and ending index of a word. For this, it will end up going through all the characters in the input string. So, the outer loop has a time complexity of O(n). Once a word boundary has been determined, the inner loop reverses all the characters in that word in place. To do this, it uses the same algorithm as you had for reverse string algorithm - which is O(n). 
Imagine a scenario where there is only one word in the input starting at index 0 and ending at index length-1. In this case, to determine the beginning and end of the word, will need n steps. After that (and hence addition), reversing the whole word would take n/2 steps. Leading to a time complexity of O(n + n/2) = O(n)
If there are only white spaces in the input, then the outer loop will take n steps and the inner loop will not get executed at all. Resulting in O(n) time complexity.
In conclusion, although you have nested loops, it's important to look closer to understand the number of operations the algorithm is taking as compared to the size or value of the input. In this case, it happens to be O(n) time complexity.

Slack me if you have further questions on this.
